compiler/rustc_hir_analysis/src/hir_ty_lowering/lint.rs

```diff
@@ -120,9 +123,6 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
                 return;
             };
             let sugg = self.add_generic_param_suggestion(generics, self_ty.span, &impl_trait_name);
-            if sugg.is_empty() {
-                return;
-            };
```

Comment:
Also removes this check in a couple of places because `add_generic_param_suggestion()` always returns a non-empty vector.